cmd/server/cmd/crd.go

```diff
@@ -74,40 +90,56 @@ func listCrdsInstances(printf shared.FormatFn, infos map[string]template.Info) {
 	for _, tmplName := range tmplNames {
 		info := infos[tmplName]
 		// TODO : Use the plural name from the template info
-		printCrd(printf, info.Name, info.Impl, info.Name+"s", "mixer-instance")
+		printCrd(printf, fatalf, info.Name, info.Impl, info.Name+"s", "mixer-instance")
 	}
 }
 
-func printCrd(printf shared.FormatFn, shrtName, implName, pluralName, label string) {
-	group := "config.istio.io"
-	crd := apiextensionsv1beta1.CustomResourceDefinition{
-		TypeMeta: meta_v1.TypeMeta{
-			Kind:       "CustomResourceDefinition",
-			APIVersion: "apiextensions.k8s.io/v1beta1",
-		},
-		ObjectMeta: meta_v1.ObjectMeta{
-			Name: pluralName + "." + group,
-			Labels: map[string]string{
-				"impl":  implName,
-				"istio": label,
-			},
-		},
-		Spec: apiextensionsv1beta1.CustomResourceDefinitionSpec{
-			Group:   group,
-			Version: "v1alpha2",
-			Scope:   apiextensionsv1beta1.NamespaceScoped,
-			Names: apiextensionsv1beta1.CustomResourceDefinitionNames{
-				Plural:   pluralName,
-				Singular: shrtName,
-				Kind:     shrtName,
-			},
-		},
+type crdVar struct {
+	ShrtName   string
+	ImplName   string
+	PluralName string
+	Label      string
+	Name       string
+	Group      string
+	Version    string
+}
+
+// newCrdVar get a new crdVar
+func newCrdVar(shrtName, implName, pluralName, label string) *crdVar {
+	return &crdVar{
+		ShrtName:   shrtName,
+		ImplName:   implName,
+		PluralName: pluralName,
+		Label:      label,
+		Name:       pluralName + "." + Group,
+		Group:      Group,
+		Version:    Version,
 	}
-	out, err := yaml.Marshal(crd)
-	if err != nil {
-		printf("%s", err)
-		return
+}
+
+func printCrd(printf, fatalf shared.FormatFn, shrtName, implName, pluralName, label string) {
```

Comment:
Curious why change from object model to text/template ? The original motivation such that in the future if we want to use the client library to publish CRD directly along with printing, the same code can be used.

Reply:
I think this is simpler code. If we decide to push it in, we can move back to using k8s.